doc/theoretical_description_conformity_scores.rst

```diff
@@ -0,0 +1,93 @@
+.. title:: Theoretical Description : contents
+
+.. _theoretical_description_conformity_scores:
+
+=========================================
+Theoretical Description Conformity Scores
+=========================================
+
+The :class:`mapie.conformity_scores.ConformityScore` class implements various
+methods to compute conformity scores for regression.
+We give here a brief theoretical description of the scores included in the module.
+Note that it is possible for the user to create any conformal scores that are not 
+already included in MAPIE by inheriting this class.
+
+Before describing the methods, let's briefly present the mathematical setting.
+With conformal predictions we want to transform an heuristic notion of uncertainty
+from a model into a rigorous one, and the first step to do it is to choose a conformal score.
+The only requirement for the score function :math:`s(X, Y) \in \mathbb{R}` is
+that larger scores should encode worse agreement between :math:`X` and :math:`Y`. [1]
+
+There are two types of scores : the symmetric and asymmetric ones.
+The symmetric property defines the way of computing the quantile of the conformity
+scores when calculating the interval's bounds. If a score is symmetrical two
+quantiles will be computed : one on the right side of the distribution
+and the other on the left side.
+
+1. The absolute residual score
+==============================
+
+The absolute residual score (:class:`mapie.conformity_scores.AbsoluteResidualScore`)
+is the simplest and most commonly used conformal score, it translates the error
+of the model : in regression it is called the residual.
+
+.. math:: |Y-\hat{\mu}(X)|
+
+With this score the intervals of predictions will be constant over the whole dataset.
+This score is symmetric (*see above for definition*).
+
+2. The gamma score
+==================
+
+The gamma score (:class:`mapie.conformity_scores.GammaConformityScore`) adds a
+notion of adaptivity with the normalization of the residuals by the predictions.
+
+.. math:: \frac{|Y-\hat{\mu}(X)|}{\hat{\mu}(X)}
+
+It computes adaptive intervals : intervals of different size on each example.
+This score is asymmetric (*see definition above*).
+
+Compared to the absolute residual score, it allows to see regions with smaller intervals
+than others which are interpreted as regions with more certainty than others.
```

Comment:
SUGGESTION: you can add more details about this adaptability. Uncertainties are proportional to the order of magnitude of the prediction (for example if $y=10$ then $\delta y = \pm 1$, if $y=100$ then $\delta y = \pm 10$). In other words, if my prediction is multiplied by 10, my uncertainties are also multiplied by 10.

Reply:
Done

Reply:
Done